pkg/operator/ceph/cluster/cluster.go

```diff
@@ -61,9 +63,17 @@ type cluster struct {
 }
 
 func newCluster(c *cephv1.CephCluster, context *clusterd.Context) *cluster {
-	return &cluster{Namespace: c.Namespace, Spec: &c.Spec, context: context,
-		stopCh:   make(chan struct{}),
-		ownerRef: ClusterOwnerRef(c.Namespace, string(c.UID))}
+	return &cluster{
+		Namespace: c.Namespace,
+		Spec:      &c.Spec,
+		context:   context,
+		stopCh:    make(chan struct{}),
+		ownerRef:  ClusterOwnerRef(c.Namespace, string(c.UID)),
+		// at this phase of the cluster creation process, the identity components of the cluster are
+		// not yet established. we reserve this struct which is filled in as soon as the cluster's
+		// identity can be established.
+		Identity: identity.BlankIdentity(),
```

Comment:
I went back and forth on this a lot myself. In the end, I reasoned that `BlankIdentity()` is more explicit than `NewIdentity()`, which might establish initial information the user doesn't know about. This is part of my desire to be extremely explicit that the identity field is empty when first created and only filled in later.

Reply:
Instead of `NewIdentity()`, I was suggesting `new(identity.Identity)`, which would give you back a pointer to an empty struct. But `BlankIdentity()` is equivalent so either way is fine.